### PR TITLE
fix(DB/Gossip): Arcanist Ithanas gossip

### LINKED ISSUE #6734: Bug(NPC): Arcanist Ithanas gossip
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [X] enUS
- [ ] other (specify)

##### FACTION

- [ ] Alliance
- [X] Horde

##### CONTENT PHASE:

- [ ] Generic
- [X] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

Original report: https://github.com/chromiecraft/chromiecraft/issues/1097

##### CURRENT BEHAVIOUR:

Missing gossip menu's for npc.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

https://wowpedia.fandom.com/wiki/Arcanist_Ithanas You can check here his gossips.

##### SOURCE:

https://wowpedia.fandom.com/wiki/Arcanist_Ithanas

##### STEPS TO REPRODUCE THE PROBLEM:

1. Complete first quest in Blood elf starting zone and take his quest afterwards.

##### EXTRA NOTES:

image

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/cecbd604d3f57312165cc53f8ed40feae51ee45f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

Confirmed and reproduced on a local AC server (AC HASH/COMMIT: dc76e6b) :

WoWScrnShot_070521_072532

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/99776030-bug-npc-arcanist-ithanas-gossip?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Zenslash**
## Changes Proposed:
-  Added new gossip for Arcanist Ithanas cause it was not in npc_text
-  Created new gossip menu cause Arcanist doesn't have one
-  Added condition for the second gossip cause it must be shown only when ```A Fistful of Slivers``` quest completed

## Issues Addressed:
- Closes #6734 
- Closes chromiecraft/chromiecraft#1097

## SOURCE:
- https://wowpedia.fandom.com/wiki/Arcanist_Ithanas

## Tests Performed:
- Tested on local server

## How to Test the Changes:
1.  Go to ```Arcanist Ithanas``` and talk with him. You will see the first gossip
2. Take and complete ```A Fistful of Slivers```
3. Return to ```Arcanist Ithanas``` and talk with him. You will see the second gossip

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1625500923667306000.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+INSERT INTO `npc_text`(`ID`,`text0_0`,`text0_1`,`BroadcastTextID0`,
+`lang0`,`Probability0`)
+VALUES
+(50031,
+"Alas $N, I am much too busy to talk. As I mentioned, I need time to think on the situation at hand.
+
+I wish you luck in your travels. Good day.",
+"Alas $N, I am much too busy to talk. As I mentioned, I need time to think on the situation at hand.
+
+I wish you luck in your travels. Good day.",0,0,0);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1625500923667306000.sql:13]: As long as this comes out correct in-game I am fine with it.

Usually I use $B for Linebreak.

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1625500923667306000.sql (hunk 1 above)
Comment (Zenslash) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql]: @Kitzunu Thanks for your review. I rewrote gossip with $B as you recommended

File: data/sql/updates/pending_db_world/rev_1625500923667306000.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/* Create new gossip menuID */
+DELETE FROM `gossip_menu` WHERE (`MenuID` = 61023) AND (`TextID` IN (15296, 50031));
+INSERT INTO `gossip_menu` (`MenuID`, `TextID`) VALUES
+(61023, 15296),
+(61023, 50031);
+
+/* Add condition for the second gossip */
+DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 14) AND (`SourceGroup` = 61023) AND (`SourceEntry` = 50031) AND (`SourceId` = 0) AND (`ElseGroup` = 0) AND (`ConditionTypeOrReference` = 28) AND (`ConditionTarget` = 0) AND (`ConditionValue1` = 8336) AND (`ConditionValue2` = 0) AND (`ConditionValue3` = 0);
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
+(14, 61023, 50031, 0, 0, 8, 0, 8336, 0, 0, 0, 0, 0, '', '');
```
[the remark this answers is not in the document]
Comment (Zenslash) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql:20]: Didn't understand a little what the problem is here

[the remark this answers is not in the document]
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql:20]: You are missing the comment =)

[the remark this answers is not in the document]
Comment (Zenslash) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql:20]: Added comment to the last line

[the remark this answers is not in the document]
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql:20]: No. see https://www.azerothcore.org/wiki/conditions#comment it is for the conditions tabel

[the remark this answers is not in the document]
Comment (Zenslash) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql:27]: @Kitzunu I modified the comment. Take a look please

[the remark this answers is not in the document]
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql:27]: Not quite. I will show you tonight when I am on my PC :)

File: data/sql/updates/pending_db_world/rev_1625500923667306000.sql [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/*
+	SourceTypeOrReferenceId = CONDITION_SOURCE_TYPE_GOSSIP_MENU = 14
+	ConditionTypeOrReference = CONDITION_QUESTREWARDED = 8
+    ConditionValue1(quest_template id) = 8336
+    ConditionValue2 - always 0
+    ConditionValue3 - always 0
+*/
+DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 14) AND (`SourceGroup` = 61023) AND (`SourceEntry` = 50031) AND (`SourceId` = 0) AND (`ElseGroup` = 0) AND (`ConditionTypeOrReference` = 8) AND (`ConditionTarget` = 0) AND (`ConditionValue1` = 8336) AND (`ConditionValue2` = 0) AND (`ConditionValue3` = 0);
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
+(14, 61023, 50031, 0, 0, 8, 0, 8336, 0, 0, 0, 0, 0, '', '');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1625500923667306000.sql:27]:
```suggestion
(14, 61023, 50031, 0, 0, 8, 0, 8336, 0, 0, 0, 0, 0, '', 'Show gossip only if quest 8336 is rewarded');
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1625500923667306000.sql (hunk 3 above)
Comment (Zenslash) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql]: @Kitzunu Thanks, Now I understand what you meant )

File: data/sql/updates/pending_db_world/rev_1625500923667306000.sql [hunk 4]
```diff
@@ -0,0 +1,30 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1625500923667306000');
+
+/* Add new gossip */
+INSERT INTO `npc_text`(`ID`,`text0_0`,`text0_1`,`BroadcastTextID0`,
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1625500923667306000.sql:4]: please use `DELETE` before `INSERT` to make your queries re-executable (whenever possible)

Comment (Zenslash) [reply data/sql/updates/pending_db_world/rev_1625500923667306000.sql:5]: @FrancescoBorzi I added ```DELETE```. Could you take a look?

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**